quasar/splunk_nova_return.py

```diff
@@ -0,0 +1,203 @@
+import socket
+
+# Imports for http event forwarder
+import requests
+import json
+import time
+
+import logging
+
+_max_content_bytes = 100000
+http_event_collector_SSL_verify = False
+http_event_collector_debug = True
+
+log = logging.getLogger(__name__)
+
+hec = None
+
+
+def returner(ret):
+    # Customized to split up the queries and extract the correct sourcetype
+
+    opts = _get_options()
+    logging.info("Options: %s" % json.dumps(opts))
+    http_event_collector_key = opts['token']
+    http_event_collector_host = opts['indexer']
+    # Set up the collector
+    hec = http_event_collector(http_event_collector_key, http_event_collector_host)
+    # st = "salt:hubble:nova"
+    data = ret['return']
+    minion_id = ret['id']
+    jid = ret['jid']
+    master = __grains__['master']
+
+    for fai in data['Failure']:
+        check_id = fai.keys()[0]
+        payload = {}
+        event = {}
+        event.update({"minion_id": minion_id})
+        event.update({"check_result": "Failure"})
+        event.update({"check_id": check_id})
+        event.update({"job_id": jid})
+        event.update({"master": master})
+        payload.update({"host": minion_id})
+        payload.update({"index": opts['index']})
+        payload.update({"sourcetype": opts['sourcetype']})
+        payload.update({'event': event})
+        hec.batchEvent(payload)
+
+    for suc in data['Success']:
+        check_id = suc.keys()[0]
+        payload = {}
+        event = {}
+        event.update({"minion_id": minion_id})
+        event.update({"check_result": "Success"})
+        event.update({"check_id": check_id})
+        event.update({"job_id": jid})
+        event.update({"master": master})
+        payload.update({"host": minion_id})
+        payload.update({"sourcetype": opts['sourcetype']})
+        payload.update({"index": opts['index']})
+        payload.update({"event": event})
+        hec.batchEvent(payload)
+
+    payload = {}
+    payload.update({"minion_id": minion_id})
+    payload.update({"compliance_percentage": ret['return']['Compliance']})
```

Comment:
Compliance can also be disabled, so maybe put this in an if statement that checks if compliance is present?